bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/Storage.java

```diff
@@ -1740,29 +1740,13 @@ private Properties findVMProfile(Generation systemGeneration) {
 
 			result = new Properties();
 			vmProfile = JAVASE + embeddedProfileName + javaSpecVersion;
-			InputStream profileIn = null;
-			if (vmProfile != null) {
-				// look for a profile in the system bundle based on the vm profile
-				String javaProfile = vmProfile + PROFILE_EXT;
-				profileIn = findInSystemBundle(systemGeneration, javaProfile);
-				if (profileIn == null)
-					profileIn = getNextBestProfile(systemGeneration, JAVASE, runtimeVersion, embeddedProfileName);
-			}
-			if (profileIn == null)
-				// the profile url is still null then use the min profile the framework can use
-				profileIn = findInSystemBundle(systemGeneration, "JavaSE-1.8.profile"); //$NON-NLS-1$
-			if (profileIn != null) {
-				try {
+			try (InputStream profileIn = createProfileStream(systemGeneration, vmProfile, embeddedProfileName)) {
+				if (profileIn != null) {
 					result.load(new BufferedInputStream(profileIn));
```

Comment:
I have the impression that Properties.load() already uses a buffer internally, therefore a BufferedInputStream would probably only add extra copying and GC work.
So result.load(profileIn); would probably be enough and even better.